src/Quizzer.API.BL/Facades/QuestionFacade.cs

```diff
@@ -0,0 +1,332 @@
+﻿// Ignore Spelling: API
+
+using AutoMapper;
+using Quizzer.API.BL.Models;
+using Quizzer.API.DAL.Entities;
+using Quizzer.API.DAL.Enums;
+using Quizzer.API.DAL.Repository.Interfaces;
+using Quizzer.API.DAL.UnitOfWork;
+using Quizzer.Extensions;
+
+namespace Quizzer.API.BL.Facades {
+    public class QuestionFacade : FacadeBase<QuestionEntity, QuestionDetailModel> {
+        public QuestionFacade(IUnitOfWorkFactory unitOfWorkFactory, IMapper mapper) : base(unitOfWorkFactory, mapper) {
+        }
+
+        protected override string[] IncludesNavigationPathDetails => new string[] {
+            $"{nameof(QuestionEntity.Answers)}" ,
+            $"{nameof(QuestionEntity.UserAnswers)}"
+        };
+
+        public override QuestionDetailModel GetDetailById(Guid id) {
+            QuestionDetailModel model = base.GetDetailById(id);
+
+            foreach (AnswerDetailModel answer in model.Answers) {
+                answer.Type = model.QuestionType;
+            }
+
+            return model;
+        }
+
+        public void AddAnswer(Guid questionId, Guid answerId) {
+            IUnitOfWork uow = p_unitOfWorkFactory.Create();
+
+            using IRepository<QuestionEntity> questionRepository = uow.GetRepository<QuestionEntity>();
+            using IRepository<AnswerEntity> answerRepository = uow.GetRepository<AnswerEntity>();
+
+            QuestionEntity question = questionRepository.GetById(questionId, nameof(QuestionEntity.Answers)) ??
+                throw new ArgumentException("Question ID is not valid");
+
+            if (question.Answers.Any(a => a.Id == answerId)) {
+                throw new ArgumentException("Answer is already in question");
+            }
+
+            AnswerEntity answer = answerRepository.GetById(answerId) ??
+                throw new ArgumentException("Answer does not exist");
+
+            if (question.Answers.Any(a => a.Order == answer.Order)) {
+                throw new ArgumentException("Order of answer must be unique");
```

Comment:
When there is a single/multichoice question, would´n this make a problem?

Reply:
Fixed and completed the function (as I for some reason did not complete the function...)

Reply:
Did you pushed that commit? It seems to me same on git and there is last commit 2 days ago.